### ISSUE TITLE
Remove DSL, and use only elixir structures.

### ISSUE THREAD

**Issue by OpakAlex** [Zatvobor/tirexs#240]

**Comment by OpakAlex** [comment]
Extract dsl to new library `tirexs-dsl`

**Comment by dfcarpenter** [comment]
I'm newish to elixir but have been using this library at work and would like to contribute. Could you give me a quick example of what you mean be using plain elixir structures instead of macros?

**Comment by OpakAlex** [comment]
json it's so easy format, like: 
`query: {term: 'a'}` => `[query: [term: 'a']]`
So i think we can remove all macroses and just do this.

**Comment by Qqwy** [comment]
One problem the current macros have, is that it is impossible to include `if` statements or `for` list comprehensions in them. This makes it impossible to e.g. only add a filter if some condition is true, or write out a list of `term`s to match (which is not the same as using an ElasticSearch `terms` query).

**Comment by Qqwy** [comment]
I don't think this feature is really necessary, because Tirexs already happily works when you give it a map or a keyword list. So you can create the Elixir equivalent of any of the JSON-examples, and Tirexs will be able to use this.
